conans/model/info.py

```diff
@@ -481,4 +486,30 @@ def shared_library_package_id(self):
             for dep_name in self.requires.pkg_names:
                 dep_options = self.full_options[dep_name]
                 if "shared" not in dep_options or not self.full_options[dep_name].shared:
-                    self.requires[dep_name].package_revision_mode()
\ No newline at end of file
+                    self.requires[dep_name].package_revision_mode()
+
+    def base_compiler_compatible(self, intel_compiler_version):
```

Comment:
I think we should aim for the follow ``package_id()``:

```python
def package_id(self):
     if self.settings.compiler == "intel":
            compatible = self.info.clone()
            compatible.base_compatible()
            ...
     elif self.settings.compiler == "Visual Studio":
           compatible = self.info.clone()
           compatible.parent_compatible(compiler="intel", version="18")
           ....
```

This means:
- ``if self.full_settings.compiler.base`` shouldn't be checked, or if checked is to return an error if "base" is not there, not silently go away.
- The compatibility methods are fully generic for any other compilers.

Reply:
Yes, very nice ideas.